web/components/groups/group-about-post.tsx

```diff
@@ -0,0 +1,143 @@
+import { useAdmin } from 'web/hooks/use-admin'
+import { Row } from '../layout/row'
+import { Content } from '../editor'
+import { TextEditor, useTextEditor } from 'web/components/editor'
+import { Button } from '../button'
+import { Spacer } from '../layout/spacer'
+import { Group } from 'common/group'
+import { deleteFieldFromGroup, updateGroup } from 'web/lib/firebase/groups'
+import PencilIcon from '@heroicons/react/solid/PencilIcon'
+import { DocumentRemoveIcon } from '@heroicons/react/solid'
+import { createPost } from 'web/lib/firebase/api'
+import { Post } from 'common/post'
+import { deletePost, updatePost } from 'web/lib/firebase/posts'
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+
+export function GroupAboutPost(props: {
+  group: Group
+  isCreator: boolean
+  post: Post
+}) {
+  const { group, isCreator, post } = props
+  const isAdmin = useAdmin()
+
+  if (group.aboutPostId == null && !isCreator) {
+    return <p className="text-center">No post has been created </p>
+  }
+
+  return (
+    <div className="rounded-md bg-white p-4">
+      {isCreator || isAdmin ? (
+        <RichEditGroupAboutPost group={group} post={post} />
+      ) : (
+        <Content content={post.content} />
+      )}
+    </div>
+  )
+}
+
+function RichEditGroupAboutPost(props: { group: Group; post: Post }) {
+  const { group, post } = props
+  const [editing, setEditing] = useState(false)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const router = useRouter()
+
+  const { editor, upload } = useTextEditor({
+    defaultValue: post.content,
+    disabled: isSubmitting,
+  })
+
+  async function savePost() {
+    if (!editor) return
+    const newPost = {
+      title: group.name,
+      content: editor.getJSON(),
+    }
+
+    if (group.aboutPostId == null) {
+      const result = await createPost(newPost).catch((e) => {
+        console.error(e)
+        return e
+      })
+      await updateGroup(group, {
+        aboutPostId: result.post.id,
+      })
+    } else {
+      await updatePost(post, {
+        content: newPost.content,
+      })
+    }
+    await router.replace(router.asPath)
```

Comment:
It's not the best solution to refresh the page. If you implement the `usePost` hook mentioned above, this might not be necessary. Then you could just set `setEditing(false)`.